index.js

```diff
@@ -48,17 +49,26 @@ function requestbody(opts) {
     var body = {};
     // so don't parse the body in strict mode
     if (!opts.strict || ["GET", "HEAD", "DELETE"].indexOf(this.method.toUpperCase()) === -1) {
-      if (opts.json && this.is('json'))  {
-        body = yield buddy.json(this, {encoding: opts.encoding, limit: opts.jsonLimit});
-      }
-      else if (opts.urlencoded && this.is('urlencoded')) {
-        body = yield buddy.form(this, {encoding: opts.encoding, limit: opts.formLimit});
-      }
-      else if (opts.text && this.is('text')) {
-        body = yield buddy.text(this, {encoding: opts.encoding, limit: opts.textLimit});
-      }
-      else if (opts.multipart && this.is('multipart')) {
-        body = yield formy(this, opts.formidable);
+      try {
+        if (opts.json && this.is('json'))  {
+          body = yield buddy.json(this, {encoding: opts.encoding, limit: opts.jsonLimit});
+        }
+        else if (opts.urlencoded && this.is('urlencoded')) {
+          body = yield buddy.form(this, {encoding: opts.encoding, limit: opts.formLimit});
+        }
+        else if (opts.text && this.is('text')) {
+          body = yield buddy.text(this, {encoding: opts.encoding, limit: opts.textLimit});
+        }
+        else if (opts.multipart && this.is('multipart')) {
+          body = yield formy(this, opts.formidable);
+        }
+        
+      } catch(parsingError) {
+        if (typeof(opts.onError) == 'function') {
+          opts.onError(parsingError, this);
```

Comment:
@dodekeract yes typeof is an operand an it can be used without round brackets.
@tunnckoCore I agree, it can be done in 2 lines but sadly there are a lot of values that javascript evaluates as truthy which will throw an error when trying to use them as functions.